source/base/mpi.cc

```diff
@@ -91,6 +91,72 @@ namespace Utilities
     }
 
 
+
+    int create_group(const MPI_Comm  &comm,
+                     const MPI_Group &group,
+                     const int        tag,
+                     MPI_Comm        *new_comm)
+    {
+#if DEAL_II_MPI_VERSION_GTE(3, 0)
+      return MPI_Comm_create_group(comm, group, tag, new_comm);
+#else
+      int rank;
+      AssertThrowMPI(MPI_Comm_rank(comm, &rank));
```

Comment:
I think this style is confusing because I don't recall whether the argument inside the `AssertThrowMPI` is executed in release mode. Please write it as
```
  int err = MPI_Comm_rank (...);
  AssertThrowMPI(err);
```
Same below.